### PR TITLE
Remove Rails.application.config.load_database_yaml and DummyConfig

### PR DESCRIPTION
**Pull request by eikes**
This change does the same thing as the DummyConfig, only directly on the `Rails.application.config.singleton_class`.

We could also define a `to_s` method for the YAML compiler to fill in dummy values, but we don't need to, because the `Object#to_s` method works too.

It is an improvement on the changes in 5ba8aa585491c6e7d0af1e6ed772cb67ced34ea7

### CODE REVIEW

**Comment by eikes** [comment]
I am not sure about the `ENV["DATABASE_URL"]` part in `#database_configuration` but I think the tests pass for this as well:

```ruby
      def load_database_yaml # :nodoc:
        Rails.application.config.singleton_class.define_method(:method_missing) { |selector, *args, &blk| self }

        database_configuration
      ensure
        Rails.application.config.singleton_class.remove_method(:method_missing)
      end
```

File: activerecord/lib/active_record/tasks/database_tasks.rb [hunk 1]
```diff
@@ -138,9 +138,14 @@ def create_all
       end
 
       def setup_initial_database_yaml
-        return {} unless defined?(Rails)
+        return {} if !defined?(Rails) || Rails.application.config.paths["config/database"].existent.empty?
 
-        Rails.application.config.load_database_yaml
+        begin
+          Rails.application.config.singleton_class.define_method(:method_missing) { |selector, *args, &blk| self }
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/tasks/database_tasks.rb:144]: I don't see why the rails framework would essentially monkey patch an object it owns. I don't think this is a good solution.

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

**Comment by rafaelfranca** [comment]
Thank you for the pull request. While the previous implementation is more verbose, it is also more clear what is happening. Temporarily monkey patching an object the framework owns is confusing and not clear enough to who is reading the code.

File: activerecord/lib/active_record/tasks/database_tasks.rb (hunk 1 above)
Comment (eikes) [reply activerecord/lib/active_record/tasks/database_tasks.rb:144]: Rails already does that, it's just more elegant to monkeypatch than to replace that object: https://github.com/rails/rails/blob/main/railties/lib/rails/application/configuration.rb#L388-L403

Comment (rafaelfranca) [reply activerecord/lib/active_record/tasks/database_tasks.rb:144]: It isn't. It is more intrusive, prone to problems and confusing in this case. Monkey patches can be elegant, but it doesn't mean they are easier to understand than swapping an object to something people can look in the implementation

Comment (eikes) [reply activerecord/lib/active_record/tasks/database_tasks.rb:144]: Fair point. I think we could still improve by using `ActiveSupport::ConfigurationFile.parse` and not doing all that in the `load_database_yaml` manually. Would you accept a PR for that?

Comment (rafaelfranca) [reply activerecord/lib/active_record/tasks/database_tasks.rb:144]: Yes. I think that makes sense.

Comment (eikes) [reply activerecord/lib/active_record/tasks/database_tasks.rb:144]: Like this? https://github.com/rails/rails/pull/46161/files

**Pull request closed without merging**